### ISSUE TITLE
Embedding sharing may cause NaN loss

### ISSUE THREAD

**Issue by TingxunShi** [OpenNMT/OpenNMT-tf#220]
Hello,

I'm training a GEC (grammartical error correction) model so the source language and target language are same. Therefore I set `share_embeddings` parameter in `onmt.modes.SequenceToSequence` to `onmt.models.EmbeddingsSharingLevel.SOURCE_TARGET_INPUT`

However, This lead to a NanLossDuringTrainingError in the very beginning stage of training (less than 100 steps). Remove the embeddings sharing level setting makes everything fine.

Please kindly help look into it. Below attached the hyperparams in config file:
```yaml
optimizer: AdamOptimizer:
  beta1: 0.8
  beta2: 0.998
learning_rate: 0.001
param_init: 0.1
clip_gradients: 5.0
beam_width: 8
length_penalty: 0.2
maximum_iterations: 250
```

ps: My OpenNMT-tf version is 1.7.0 and TF version is 1.10.1

**Comment by guillaumekln** [comment]
Hi,

Sharing embeddings means gradients are accumulated, which could cause issues with some optimization settings. Could you first try with a lower learning rate?

**Comment by TingxunShi** [comment]
Hi,

Thanks for the advice. I changed the learning rate to 1e-7 and the NaN problem seems has gone, however the convergence speed is also quite slow (but could be foreseen)

I'll come back to close this issue when I confirm the network converges. Many thanks!

**Comment by guillaumekln** [comment]
I don't think you need to go that far in lowering the learning rate. Did you try with something like 0.0002?

**Comment by TingxunShi** [comment]
Seems I was too radical, 0.0001 is good enough. Thanks for the help!
